traits/src/grid.rs

```diff
@@ -1,13 +1,15 @@
 //! Geometry and topology definitions
 
-pub use crate::cell::ReferenceCell;
-pub use crate::element::FiniteElement;
+use solvers_tools::arrays::AdjacencyList;
 
 pub trait Geometry {
-    fn reference_cell(&self) -> &dyn ReferenceCell;
-    fn map(&self, reference_coords: &[f64], physical_coords: &mut [f64]);
-
     fn dim(&self) -> usize;
+
+    fn point(&self, i: usize) -> Option<&[f64]>;
+
+    unsafe fn point_unchecked(&self, i: usize) -> &[f64];
```

Comment:
Do we really need this unchecked method? It is not like fast iteration over individual array elements where you want each access to be unchecked. Or is it?

Reply:
probably don't need unchecked. I've removed them for now. Can always add them back later if we find we need them